pkgs/development/libraries/cdo/default.nix

```diff
@@ -0,0 +1,60 @@
+{ stdenv, fetchurl, curl
+, enable_data ? true        # DATA support [default=yes]
+, enable_grib ? true        # GRIB support [default=yes]
+, enable_cgribex ? true     # Use the CGRIBEX library [default=yes]
+, enable_service ? true     # Use the service library [default=yes]
+, enable_extra ? true       # Use the extra library [default=yes]
+, enable_ieg ? true         # Use the ieg library [default=yes]
+, enable_cdi_lib ? false    # build, install and link to a CDI library [default=no]
+, enable_all_static ? false # build a completely statically linked CDO binary  [default=no]
+, enable_cxx ? false        # Use CXX as default compiler [default=no]
+, with_gnu_ld ? false       # assume the C compiler uses GNU ld [default=no]
+, with_hdf5 ? true          #<yes|no|directory> (default=no) location of HDF5 library
+, hdf5 ? null
+, with_netcdf ? true        #<yes|no|directory> (default=no)location of NetCDF library
+, netcdf ? null 
+}:
+
+assert with_netcdf   -> ( netcdf != null );
+assert with_hdf5     -> ( hdf5 != null );
+
+stdenv.mkDerivation rec {
+  version = "1.7.2";
+  name = "cdo-${version}";
+  
+  # Dependencies 
+  buildInputs = [ curl ]
+    ++ stdenv.lib.optional with_netcdf [ netcdf ]
+    ++ stdenv.lib.optional with_hdf5 [ hdf5 ];
+
+  src = fetchurl {
+    url = "https://code.zmaw.de/attachments/download/12760/${name}.tar.gz";
+    sha256 = "4c43eba7a95f77457bfe0d30fb82382b3b5f2b0cf90aca6f0f0a008f6cc7e697";
+  };
+
+  # Configure phase 
+  configureFlags = ''
+    ${if enable_data then       "--enable-data" else ""}
```

Comment:
Please consider using `optionalString` or specifying `configureFlags` as a list of strings, using `optional(s)` for optional flags.